bin/sbang

```diff
@@ -56,70 +58,132 @@
 #     2    <?php #/long/path/to/php with arguments ?>
 #     3    <?php echo "success!\n"; ?>
 #
 # How it works
-# -----------------------------
-# `sbang` is a very simple bash script. It looks at the first two
-# lines of a script argument and runs the last line starting with
-# `#!`, with the script as an argument. It also forwards arguments.
+# ------------
+# `sbang` is a very simple posix shell script. It looks at the first two
+# lines of a script argument and runs the last line starting with `#!`,
+# with the script as an argument. It also forwards arguments.
 #
 
+# We disable two shellcheck errors below:
+# SC2124: when saving arguments, we intentionally assign as an array
+# SC2086: when splitting $shebang_line and exec args, we want to expand args
+
+# Generic error handling
+die() {
+    echo "$@" 1>&2;
+    exit 1
+}
+
+# set SBANG_DEBUG to make the script print what would normally be executed.
+exec="exec"
+if [ -n "${SBANG_DEBUG}" ]; then
+    exec="echo "
+fi
+
 # First argument is the script we want to actually run.
 script="$1"
 
+# ensure that the script actually exists
+if [ -z "$script" ]; then
+    die "error: sbang requires exactly one argument"
+elif [ ! -f "$script" ]; then
+    die "$script: no such file or directory"
+fi
+
 # Search the first two lines of script for interpreters.
 lines=0
-while read line && ((lines < 2)) ; do
-    if [[ "$line" = '#!'* ]]; then
-        interpreter="${line#\#!}"
-    elif [[ "$line" = '//!'*node* ]]; then
-        interpreter="${line#//!}"
-    elif [[ "$line" = '--!'*lua* ]]; then
-        interpreter="${line#--!}"
-    elif [[ "$line" = '<?php #!'*php* ]]; then
-        interpreter="${line#<?php\ \#!}"
-        interpreter="${interpreter%\ ?>}"
+while read -r line && [ $lines -ne 2 ]; do
+    if [ "${line#\#!}" != "$line" ]; then
+        shebang_line="${line#\#!}"
+    elif [ "${line#//!}" != "$line" ]; then      # // comments
+        shebang_line="${line#//!}"
+    elif [ "${line#--!}" != "$line" ]; then      # -- lua comments
+        shebang_line="${line#--!}"
+    elif [ "${line#<?php\ }" != "$line" ]; then  # php comments
+        shebang_line="${line#<?php\ \#!}"
+        shebang_line="${shebang_line%\ ?>}"
     fi
     lines=$((lines+1))
 done < "$script"
-# this is ineeded for scripts with sbang parameter
-# like ones in intltool
+
+# shellcheck disable=SC2124
+# this saves arguments for later and intentionally assigns as an array
+args="$@"
+
+# handle scripts with sbang parameters, e.g.:
 # #!/<spack-long-path>/perl -w
+<<<<<<< HEAD
 # this is the interpreter line with all the parameters as a vector
 interpreter_v=(${interpreter})
 # this is the single interpreter path
 interpreter_f="${interpreter_v[0]}"
+=======
+#
+# put the shebang line with all the parameters in the $@ array
+# and get the first element.
+# shellcheck disable=SC2086
+set $shebang_line
+set -- "$@"
+interpreter="$1"
+arg1="$2"
+>>>>>>> 3e32a32333... sbang: convert sbang script to POSIX shell
```

Comment:
I have the impression something went wrong with a conflict...

Reply:
fixed